src/os/bluestore/BlueStore.cc

```diff
@@ -8666,21 +8664,15 @@ void BlueStore::_deferred_submit_unlock(OpSequencer *osr)
 void BlueStore::_deferred_aio_finish(OpSequencer *osr)
 {
   dout(10) << __func__ << " osr " << osr << dendl;
-  assert(osr->deferred_running);
-  DeferredBatch *b = osr->deferred_running;
+  DeferredBatch *b;
 
   {
     std::lock_guard<std::mutex> l(deferred_lock);
-    assert(osr->deferred_running == b);
-    osr->deferred_running = nullptr;
-    if (!osr->deferred_pending) {
+    b = osr->deferred_running.front();
```

Comment:
I think the root cause is throttle.put don't make waiter call deferred_try_submit again.
In queue_transaction
while (!throttle_deferred_bytes.get_or_fail(txc->cost)) {
      deferred_try_submit();
      throttle_deferred_bytes.wait();
    }
I'm not sure those code whether fix this bug.